core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -219,6 +220,9 @@ class HadoopRDD[K, V](
       val bytesReadCallback = if (split.inputSplit.value.isInstanceOf[FileSplit]) {
         SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
           split.inputSplit.value.asInstanceOf[FileSplit].getPath, jobConf)
+      } else if (split.inputSplit.value.isInstanceOf[CombineFileSplit]) {
+        SparkHadoopUtil.get.getFSBytesReadOnThreadCallback(
+          split.inputSplit.value.asInstanceOf[CombineFileSplit].getPath(0), jobConf)
```

Comment:
Can you push this logic down to the SparkHadoopUtil so that we don't duplicate it in two places (HadoopRDD and NewHadoopRDD).